src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4529,3 +4529,143 @@ void C2_MacroAssembler::vector_maskall_operation32(KRegister dst, Register src,
   kunpckdql(dst, tmp, tmp);
 }
 #endif
+
+void C2_MacroAssembler::vector_reverse_bit(BasicType bt, XMMRegister dst, XMMRegister src, XMMRegister xtmp1,
+                                           XMMRegister xtmp2, XMMRegister xtmp3, Register rtmp, int vec_enc) {
+  if (VM_Version::supports_avx512bw()) {
```

Comment:
`VM_Version::supports_avx512bw() && (VM_Version::supports_avx512vl() || vec_enc == AVX_512bit)`

Reply:
VPSHUFB used here needs AVX512VLBW feature in-order to operate over 128/256/512 bit vector. Thus for an EVEX encoded instruction limiting feature here is AVX512VLBW.